### ISSUE TITLE
Misc Issues / Feedback from Kyle

### ISSUE THREAD

**Issue by kwiens** [iFixit/ifixit-microsoft#6]
Critical:
- [x] Caution bullets are not shown (special icons are currently blue). It's safety critical that we show caution and note bullets somehow.
- [x] Step line text needs to render simple HTML (links, bold, italic, etc.). See step 7 of the orange teardown for example.
- [x] Numerous crashes browsing guides, switching between social / speech, account, search, and the guide. Emailed crash reports.
- [x] On login: enter email adress but no password. Hit √ to login. App crashes. I got several crashes around the login screen, and eventually had to completely restart the app to be able to log in.
- [x] Login kept failing to log me in (error message), but my profile info still showed. Eventually after restarting the app, it worked. It seems that login only works if it's the very first thing you try to do.
- [x] Open with VS2012 the file translation.resx and check for english / typos etc etc. and send me back the file.
- [x] User account creation is broken with CSRF token failure.
- [x] Adding a guide to favorites takes an unreasonably long amount of time.
- [x] Visiting a guide twice doesn't work. Ex: open a guide, back out of it, and try to re-enter it.
- [x] If you have no internet, it warns you, but still lets you try to access stuff. Which causes views to crash.

Bugs:
- [x] On missions page there appears to be missing headers or odd spacing
- [x] On support tab add question marks to the header questions that do not have them.
- [x] When trying to login with empty fields, you get the error message "please fill the form completely". Change this to "Please fill out all fields."
- [x] On the Featured tab, there's blank / empty space in the top of the box that shows the most recent collection (Popular Tablet Repairs). It looks like the content of that box is just pushed down too far.
- [x] On the support tab, "repository" is misspelled.
- [x] On guide load, it flashes 'No data to display' briefly before the guide loads
- [x] 'Prerequisites' is shown even if there are none. I don't think prerequisites need to be shown at all. If you show them, call it 'Sections' and don't make them hyperlinks. The prerequisite steps are already included in the guide.
- [x] Proof error messages for proper grammar and spelling. Example "please fill the form completly" -> "Please completely fill out the form"
- [x] When you swipe to a step that has multiple images, the small thumbnails flicker briefly
- [x] The scroll position is preserved across steps. That is inconvenient if you read one step completely, scrolling down, then want to go to the next step to read it from the beginning. There is no need to preserve scroll position after you swipe off a step—you can always default to showing the top of the step.
- [x] I had some guides spin showing 'Loading…' forever. Example: HP photosmart 935 digital camera batteries.
- [x] There's no way to make secondary guide photos large. They should all be large and then users will need to scroll down to see them. 
- [x] In guide view, the search button can require two taps to activate. I'm not sure what triggers this, perhaps scrolling?
- [x] There's no good reason to show the header (account + logo + search) on guide view. Can we just show the guide?
- [x] Sometimes photos from other steps appear on top of the 'correct' photo just momentarily.
- [x] The user account creation page completely froze when I used the 'peek at password' feature. Haven't been able to reproduce the error.
- [x] The user account creation page doesn't work in landscape (there is no landscape layout).
- [x] After account creation fails, it still tries to log in with the invalid credentials.
- [x] On account creation, the form asks for "username". On the site, we simply ask for "name" and suggest they use their real name. The app should follow the site’s format.
- [x] The Support page item ordering is odd. Let's reorder them as iFixit Answers, Dozuki, iFixit Meta, Github. Also, scrolling on this page doesn't work in landscape mode. 
- [x] On the login screen, the keyboard’s return key doesn't seem to do much. Instead you need to dismiss the keyboard and click the Login button.
- [x] The user does not stay logged in across app uses.
- [x] If you don't enter anything into the login boxes, it still tries to login when you click login. This will be a definite failure, so we should check locally in the app to make sure <something> has been entered in both fields before we attempt a login.
- [x] It's really not easy to tap the profile button. Nor the search button. Increase the hit box size?
- [x] If you attempt to scroll on a device page overview, the image will scroll up/down with your finger and get cut off.
- [x] When you swipe from one guide step to another, the scroll location is inherited from the previous view.
- [x] Tapping photos on guide steps doesn't magnify it, display a larger version of the image, or make it more visible.
- [x] On category pages, the category title bar overlaps title elements. If you click a level deeper and then back out, it corrects itself.
- [x] You can view an empty 'profile' page by logging in, navigating to your profile, clicking 'logout' (is 'logout' a word? Is it 'log out'?), and when it takes you back to the categories page, tapping the '<--' back button. 
- [x] On the 'news' page before content loads, the text briefly says 'no data to display' (much like the resolved issue a few bullets above).
- [x] Some of the items on the 'support' page are missing punctuation. `Want To Discuss iFixit things` -> `Want to discuss iFixit things?`. The casing is also inconsistent. Some are title case while others are sentence case, and others are a combination of the two.
- [x] On certain search result pages under the 'device' category, you can infinitely lazy-load results when scrolling down. For instance, searching for 'Apple' and scrolling down will cause the same search results to be loaded and then appended to the page for as long as you like.
- [x] On some device searches, the lazy-load _will_ stop. Though it does, the ['loading'](https://f.cloud.github.com/assets/1192362/799503/3cf26032-ed76-11e2-97fd-6f5b6c3b3946.jpg) text hangs around at the bottom of the page.
- [ ] In the news tab the text should be truncated so it doesn't extend off the edge of the screen. (see: the date the article was published)
- [x] When doing a search and no results are returned display a message to the user
- [x] On the login screen, when the phone is in landscape orientation, you can't scroll down to see the password box. Similar to next bug.
- [x] The account creation screen doesn't allow you to scroll down and enter text into the last field (confirm password), instead it will automatically scroll back up to show the first 3 fields with the keyboard covering up the last field.
- [ ] Images do some strange funky jump thingy when sliding between guide steps. They appear to stretch inside of their frame to distort, and then return to their proper place.
- [ ] Elements with no images show the ruler image, though it's not centered / correctly sized. Mostly on Device pages, [like the Netgear WGR4v9.](https://f.cloud.github.com/assets/1192362/789408/9fdae1be-eb10-11e2-9605-5071e3c4d027.jpg)

Nice-To-Haves
- [ ] On the Mission tab, add an italicized header called "Guide Creation" above the "Writing Guides" entry.
- [ ] On the Mission tab, add an italicized header called "Help" above the "Administration and Moderation Help" entry.
- [ ] When a user's Favorites list is empty, nothing is displayed. We should display some text that says "No Guides favorited yet."
- [ ] Can we show the user "About me"/summary stuff on the profile page? Maybe reputation? Owned guides? Etc...
- [ ] Device pages have no landscape layout
- [ ] Let's replace the content on the Mission tab with the content on http://www.ifixit.com/Info/what

**Comment by kwiens** [comment]
Hey @cullenarroyo, what do you think of this remaining to-do list? Should we close it out?

We have to push an updated app soon to replace a deprecated API (the code is already done for this).

**Comment by cullenarroyo** [comment]
I dont think that any of the remaining list items were glaring issues. Some, like "Device pages have not landscape layout" have probably already been fixed.

**Comment by kwiens** [comment]
I don't have a device that I can check these things against. Can you find me and we'll go through the checklist on it?
